### PR TITLE
Add man page to crystal

### PR DESCRIPTION
**Pull request by werner**

### CODE REVIEW

**Comment by asterite** [comment]
I know nothing about man pages, so if somebody else can approve this, go ahead :-)

I guess if we accept this we should also add this somewhere in the installation process so that it's available, right?

**Comment by ysbaddaden** [comment]
Just run `man ./crystal.1` to see how it renders.

File: man/crystal.1 [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+.Pp
+Show version.
+.El
+.Sh SEE ALSO
+.Bl -hang -compact -width "https://github.com/manastech/crystal/1234"
+.It http://crystal-lang.org/
+The official web site.
+.It https://github.com/manastech/crystal
+Official Repository.
+.Sh Contributing
```
Comment (ysbaddaden) [inline man/crystal.1]: I think we can skip "contributing" in the man page. Let's just stick to explain the command.

File: man/crystal.1 [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+.Nm
+.Op command
+.Op switches
+.Op program file
+.Op --
+.Op arguments
+.Sh DESCRIPTION
+Crystal is a programming language with the following goals:
+
+.Bl -bullet -offset indent -compact
```
Comment (ysbaddaden) [inline man/crystal.1]: Maybe skip the language goals, but quickly explain that `crystal` is the compiler for the Crystal language and stick to usage: explaining why `--` is required to pass arguments to the program in the `run` command, that the `run` command can just be skipped, or that one should use the `--release` flag to turn LLVM optimizations on for distributing a binary or running benchmarks.

**Comment by ysbaddaden** [comment]
Good job overall. Some more comments, about the design:

- The indentation is large in a terminal (it quickly goes up to 30 chars). It would be nice to be closer to 8, to match the Linux man pages.
- Maybe differentiate `commands` from `options`: first having a flat list of all commands (without options), then each command and their options? I'm not sure, thought, and open to suggestions :-)

**Comment by miketheman** [comment]
Very cool!

Is the intent for the man page to cover _all_ `crystal` CLI commands? If so, then the addition of `play` and others might be needed.

**Comment by miketheman** [comment]
Superseded by #2989, recommend closing this one.

**Comment by matiasgarciaisaia** [comment]
Closing in favour of #2989 - thanks @miketheman for kickstarting 👍

**Pull request closed without merging**